### DISCUSSION TITLE
'whisper' is not recognized as an internal or external command, operable program or batch file.

### DISCUSSION THREAD

**Discussion by Primuh** [openai/whisper#531] [Q&A]
Hi, I am very new to coding/computer science in general, so I realize I'm probably doing something fundamentally wrong. After running `pip install git+https://github.com/openai/whisper.git`, when I try to run any whisper command (even whisper --help), I get: 'whisper' is not recognized as an internal or external command, operable program or batch file. 

I know this is a very general and basic error, but I have done some research and troubleshooting; such as moving the path of the 'whisper' package, adding the path to Windows environment variables, or deleting and reinstalling it. Nothing I do seems to make command prompt recognize the whisper command.

I have Python 3.10.8 and all the required dependencies installed. Any help with this would be much appreciated!

**Comment by Pixeptron-Vision** [comment]
@Primuh If you have already installed whisper using  `pip install git+https://github.com/openai/whisper.git `  and got this error, you probably have to upgrade to the latest version of repo using the command given in the readme. The following resolved the problem for me:
`pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git`

**Reply by ChristophKrause** [reply]
Same problem here:

**Reply by vovchisko** [reply]
It worked for me. Thanks! 🙇 

People often forget to run the power shell with admin permission on Windows while following installation steps.

**Reply by eviL-ivan** [reply]
maybe helpfull: rerun powersherr after installation :)

**Reply by Ksandesh7** [reply]
same problem and it worked for me too. Thanks

**Comment by FurkanGozukara** [comment]
I have a video that explains how to use whisper it may help. i install on windows from scratch and run it

https://www.youtube.com/watch?v=msj3wuYf3d8

**Comment by Ahmed-Elswerky** [comment]
if anyone facing this
adding this path to environment path worked

'C:\Users\\<USER_NAME>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts'

**Reply by ragnarheil** [reply]
@Ahmed-Elswerky excellent, helped immediately after re-opening PowerShell

**Reply by USR5** [reply]
I log in specifically for you to give you a thumbs-up!

**Reply by Ahmed-Elswerky** [reply]
> I log in specifically for you to give you a thumbs-up!

Glad I helped 😊

**Reply by Diegolaquino** [reply]
Thank you very much. In my case, i put python3.11 path.

**Comment by mysteriousLUCA** [comment]
I have installed whisper using the command "pip3 install setuptools-rust" but whisper is not working. I am getting "'wispher' is not recognized as an internal or external command, operable program or batch file." again and again and then when i use the command "pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git" i am getting

ERROR: Error [WinError 2] The system cannot find the file specified while executing command git version
ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?

With the help of this 
'C:\Users\<USER_NAME>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts'

also i am not able to resolve the issue. Can some one please help me with this.

**Reply by FurkanGozukara** [reply]
have you tried my tutorial ? https://www.youtube.com/watch?v=msj3wuYf3d8

**Reply by Ahmed-Elswerky** [reply]
> I have installed whisper using the command "pip3 install setuptools-rust" but whisper is not working. I am getting "'wispher' is not recognized as an internal or external command, operable program or batch file." again and again and then when i use the command "pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git" i am getting
> 
> ERROR: Error [WinError 2] The system cannot find the file specified while executing command git version
> ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?
> 
> With the help of this 
> 'C:\Users\<USER_NAME>\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts'
> 
> also i am not able to resolve the issue. Can some one please help me with this.
> 
> 

You can install git and try again
https://git-scm.com/downloads

**Reply by mysteriousLUCA** [reply]
Its not working still

**Comment by airpiratex** [comment]
hello whisper is there a method to download in python 3.11?

**Reply by Tex2002ans** [reply]
Whisper does not support Python 3.11 yet. See:

- https://github.com/openai/whisper/pull/1171

Right now, the documentation also says whisper's:

> [...] codebase is expected to be **compatible with Python 3.8-3.10**.

Some of the dependencies (like `numba`) need to be upgraded first, _then_ Whisper can be upgraded to support 3.11 too.

**Comment by USR5** [comment]
hi,good morning guy, i meet a new problem that i wanna use my amd gpu 5700xt to run whisper,
but pytorch's website says that amd gpu is not supported on windows only on linux(pictures below),but there are two apps that can run on windows,let me show you below
https://pytorch.org/
https://github.com/sakura6264/WhisperDesktop
https://github.com/tigros/Whisperer/releases/tag/2.4


this two GUI APP can use my amd gpu 5700xt on whisper, but there are some problems,on the beginning, the subtitles is right, later,the subtitles is repeatable, so i want to try to use console to do it again,
but the problems only cpu worked in console...so i wonder why i can solve this problem, thank you.








&nbsp;
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "openai/whisper"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年4月18日(星期二) 下午4:46
***@***.***&gt;;
抄送:&nbsp;" ***@***.******@***.***&gt;;
主题:&nbsp;Re: [openai/whisper] 'whisper' is not recognized as an internal or external command, operable program or batch file. (Discussion #531)





  
I log in specifically for you to give you a thumbs-up!
  
Glad I helped 😊

**Comment by a-sadegh63** [comment]
The "whisper" command does not work on my Ubuntu system. The following code solves my problem
`export PATH=$PATH:/home/your_name/.local/bin`

**Reply by Yuanyuan-888** [reply]
Works for me!!!!!! I am also using linux system :D

**Reply by tanynova99** [reply]
Hi! It helped me too, thank you! Could you explain what you did to a newbie?

**Reply by AsmaaShAli** [reply]
@a-sadegh63 
execellent!! works now with me 
big thanks
